Packages/Tracking/Core/Runtime/Scripts/Utils/HandUtils.cs

```diff
@@ -46,9 +48,23 @@ private static void AssignBestLeapProvider()
                     s_provider = UnityEngine.Object.FindAnyObjectByType<XRLeapProviderManager>();
                     if (s_provider == null)
                     {
-                        s_provider = UnityEngine.Object.FindAnyObjectByType<LeapProvider>();
-                        if (s_provider == null)
+                        var candidates = UnityEngine.Object.FindObjectsByType<LeapProvider>(FindObjectsSortMode.None);
+                        if (candidates.Any())
```

Comment:
Might be worth use `!candidates.isEmpty()` or similar to avoid pulling in Linq? I was under the impression that generally you want to avoid Linq due to speed?

Reply:
Yes, generally so. Some lambdas also allocate. I'll change it.

Reply:
Changed this and other code to be non linq